src/renderer/components/item-object-list/item-list-layout.tsx

```diff
@@ -136,14 +138,17 @@ export class ItemListLayout extends React.Component<ItemListLayoutProps> {
   }
 
   async componentDidMount() {
-    const { isConfigurable, tableId, preloadStores } = this.props;
+    const { isConfigurable, hiddenColumns, tableId, preloadStores } = this.props;
 
     if (isConfigurable && !tableId) {
       throw new Error("[ItemListLayout]: configurable list require props.tableId to be specified");
     }
 
     if (isConfigurable && !UserStore.getInstance().hiddenTableColumns.has(tableId)) {
       UserStore.getInstance().hiddenTableColumns.set(tableId, new ObservableToggleSet());
+      hiddenColumns.forEach(columnId => (
+        UserStore.getInstance().toggleTableColumnVisibility(tableId, columnId))
+      );
```

Comment:
This is incorrect, it will toggle the columns visibility on ever mount

Reply:
Ohhhh I realize my mistake. Could you advise where I should put this logic so that it executes only once?

Reply:
I just re-checked, and it should only execute once when there is no `hiddenTableColumns` instance associated to the tableId.

Reply:
But then there will be on the next time it is rendered.

Reply:
Let me re-test that, but I assumed that the observable was linked to the tableId once on the first mount, then it would remain for the lifetime of the app since UserStore is a singleton.

Reply:
Okay so it is slightly more complicated. If you reset your hidden tables to "default" (which is currently `[]`) then the next time it is round-tripped through the persistence layer that `tableId` will be removed.

Reply:
If we do this correctly, we are able to change the defaults

Reply:
Ok I see! Mmmm I'll need to think about it a bit then If you have any insight, Im all ears.

I'm moving it to a draft to work on it a bit more.

Reply:
The proper place for this is in the `src/common/user-store/preference-helpers.ts` file. I guess.

Reply:
Thanks, I'll look into it, appreciated!